### PR TITLE
SDIT-1289 Remove SYSTEM_USER from offender curfew endpoints

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/curfews/OffenderCurfewService.java [hunk 1]
```diff
@@ -124,31 +124,26 @@ private Set<Long> curfewBookingIds(Stream<OffenderCurfew> ocs) {
         return ocs.map(OffenderCurfew::getOffenderBookId).collect(toSet());
     }
 
-    @VerifyOffenderAccess(overrideRoles = {"SYSTEM_USER", "VIEW_PRISONER_DATA"})
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/curfews/OffenderCurfewService.java:127]: This annotation wasn't doing anything as it was VerifyBookingAccess that was needed

File: src/main/java/uk/gov/justice/hmpps/prison/service/curfews/OffenderCurfewService.java [hunk 2]
```diff
@@ -124,31 +124,26 @@ private Set<Long> curfewBookingIds(Stream<OffenderCurfew> ocs) {
...
     public HomeDetentionCurfew getLatestHomeDetentionCurfew(final long bookingId) {
         return offenderCurfewRepository.getLatestHomeDetentionCurfew(bookingId, StatusTrackingCodes.REFUSED_REASON_CODES)
                 .orElseThrow(() -> new EntityNotFoundException("No 'latest' Home Detention Curfew found for bookingId " + bookingId));
     }
 
-    @VerifyOffenderAccess(overrideRoles = {"SYSTEM_USER", "VIEW_PRISONER_DATA"})
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/curfews/OffenderCurfewService.java:133]: again, this annotation wasn't doing anything

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderSentenceResource.java [hunk 3]
```diff
@@ -110,6 +113,7 @@ public HomeDetentionCurfew getLatestHomeDetentionCurfew(@PathVariable("bookingId
     @Operation(summary = "Retrieve the latest Home Detention Curfew status for a list of offender booking identifiers")
     @PostMapping("/home-detention-curfews/latest")
     @SlowReportQuery
+    @PreAuthorize("hasAnyRole('VIEW_PRISONER_DATA')")
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderSentenceResource.java:116]: Only called by client creds (no user in context)

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderSentenceResource.java [hunk 4]
```diff
@@ -99,6 +101,7 @@ public List<OffenderSentenceCalc<BaseSentenceCalcDates>> getOffenderSentencesHom
     })
     @Operation(summary = "Retrieve the current state of the latest Home Detention Curfew for a booking")
     @GetMapping("/booking/{bookingId}/home-detention-curfews/latest")
+    @VerifyBookingAccess(overrideRoles = "VIEW_PRISONER_DATA")
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderSentenceResource.java:104]: only called by auth code - but have left alternative override role in to match the other

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderSentenceResourceImplIntTest.kt [hunk 5]
```diff
@@ -455,280 +454,5 @@ class OffenderSentenceResourceImplIntTest : ResourceTest() {
     }
   }
 
-  @Nested
```
Comment (karenmillermoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderSentenceResourceImplIntTest.kt:458]: move hdc tests into separate file

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**